rj_gameplay/rj_gameplay/tactic/wall_tactic.py

```diff
@@ -0,0 +1,191 @@
+"""Tactic to build a wall between mark pt (e.g. ball) and defense pt (e.g. goal)."""
+
+from dataclasses import dataclass
+from typing import List, Optional
+from typing import Dict, Generic, List, Optional, Tuple, Type, TypeVar
+
+import stp.action as action
+import stp.rc as rc
+import stp.tactic as tactic
+import stp.role as role
+
+import rj_gameplay.eval
+import rj_gameplay.skill as skills
+from rj_gameplay.skill import move
+import stp.skill as skill
+import numpy as np
+# TODO: replace w/ global param server
+from stp.utils.constants import RobotConstants, BallConstants
+
+class wall_cost(role.CostFn):
+    """Cost function for role request.
+    """
+    def __call__(
+        self,
+        robot: rc.Robot,
+        prev_result: Optional["RoleResult"],
+        world_state: rc.WorldState
+    ) -> float:
+
+        # TODO: make closest robots form wall, rather than setting on init
+        # aka actually use this method
+        # can do this by creating mult cost instances, adding attr like "wall_pt" to this class, 
+        # and giving each roleRequest the correct cost in tactic below
+        return 0.0
+
+def find_wall_pts(num_robots: int, ball_pt: np.ndarray, goal_pt: np.ndarray, world_state: rc.WorldState) -> List[np.ndarray]:
+    """Calculates num_robots points to form a wall between the ball and goal.
+    :return list of wall_pts (as numpy arrays)
+    """
+    # TODO: param server this const
+    WALL_SPACING = BallConstants.RADIUS * 1.9
```

Comment:
Could we make this a little smaller? Also, fix the TODO please. This can be a parameter now.

Reply:
#1666